Dockerfile.integration-test

```
@@ -0,0 +1,143 @@
+
+# Install Go 1.23 from official source
+RUN wget https://go.dev/dl/go1.23.6.linux-amd64.tar.gz && \
+    tar -C /usr/local -xzf go1.23.6.linux-amd64.tar.gz && \
+    rm go1.23.6.linux-amd64.tar.gz
+
+# Set Go environment variables
+ENV PATH="/usr/local/go/bin:${PATH}"
+ENV GOROOT="/usr/local/go"
+
+# Initialize PostgreSQL database with UTF8 encoding
+RUN mkdir -p /var/lib/postgresql/data /var/run/postgresql && \
```

Hadolint (2.12.0)

[error] Do not use sudo as it leads to unpredictable behavior. Use a tool like gosu to enforce root

(DL3004)